### ISSUE TITLE
TP, TO API should allow for 'delete' of users

### ISSUE THREAD

**Issue by mtorluemke** [apache/incubator-trafficcontrol#2017]
Currently, CRU is supported, but not D.

**Comment by dangogh** [comment]
we've talked about this in the past,  but thought might be problematic and that setting a user's role to `disabled` was sufficient.

Would it need to cascade to the log,  job, and deliveryservice_request tables?

**Comment by elsloo** [comment]
@mitchell852 suggested adding a column, say, "active," on the user table to disable a user. We could then switch to a view for some APIs that return users to exclude any disabled users, while allowing us to retain changelog history associated with the inactive user. For example, the API that performs authentication needs to find a user; if that API uses the view, any inactive user that attempts to authenticate would be rejected due to not being found in the view.

Just some ideas, but I do like the idea of retaining the changelog history and any other ancillary fingerprints that might exist elsewhere in the system.

**Comment by rob05c** [comment]
+1 on "disabled". We could even call it "delete user" in the UI, and have an "undelete" button.

_Someone_ will say "I accidentally deleted my user", or delete a user with a pending DS Request they shouldn't, or something. IMO it's always better to be able to undo.

Besides, undo is better than confirmation, for UI. If users can still log in (or password reset) and "undelete", we don't need an annoying "Are you sure?" dialog. Which is just a better UI.

Less work for us too, we don't get people calling saying "I accidentally deleted my user, please fix it for me"; they can fix it themselves.

**Comment by dangogh** [comment]
+1 on disabled,  but flag should be positive, e.g. "active" -- the DELETE api endpoint would just set that.

-1 on "undo" -- yes it's nicer from a UI perspective and is easy in this particular case,  but implies that an undo should be available in other cases, e.g. editing a server's domain name.  Do we really want to introduce keeping track of changes made so they can be undone?  Confirmation should be sufficient to prevent accidental deletion.

**Comment by rob05c** [comment]
I don't agree un-deleting a user would require supporting undo history of all data fields. I agree, that would be unreasonable.

**Comment by mitchell852** [comment]
@moltzaum is going to look at this

**Comment by mitchell852** [comment]
^^ I guess the "delete" was never addressed in that PR #2834
